hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/Job.groovy

```diff
@@ -330,6 +333,23 @@ class Job {
   }
 
   /**
+   * DSL method to specify required parameters for a job.
+   * Certain parameters must be set explicitly in oder for the job to run correctly in production.
+   * Specifying those parameters here provides a clear and explicit expectation for the users of the job.
+   * For example, specifying "required parameters" in a job template,
+   * and set those parameters in a job cloned from this template.
+   * This helps avoid production errors in an early stage, such as those caused by default parameters.
+   *
+   * @param args Args whose key 'parameters' has a list of value specifying the required parameters
+   */
+  @HadoopDslMethod
+  void required(Map args) {
+    LinkedHashSet<String> parameters = args.parameters;
```

Comment:
@jianlingzhong this is something that users set - I don't think that args.parameters will come out as a LinkedHashSet. Users should pass it in as a List<String> or something similar, no?